providers/fcm/src/lib/fcm.provider.ts

```diff
@@ -0,0 +1,47 @@
+import {
+  ChannelTypeEnum,
+  ISendMessageSuccessResponse,
+  IPushOptions,
+  IPushProvider,
+} from '@novu/stateless';
+import { initializeApp, cert } from 'firebase-admin/app';
+import { getMessaging, Messaging } from 'firebase-admin/messaging';
+import crypto from 'crypto';
+
+export class FcmPushProvider implements IPushProvider {
+  id = 'fcm';
+  channelType = ChannelTypeEnum.PUSH as ChannelTypeEnum.PUSH;
+
+  private messaging: Messaging;
+  constructor(
+    private config: {
+      secretKey: string;
+    }
+  ) {
+    this.config = config;
+    const firebase = initializeApp(
+      {
+        credential: cert(JSON.parse(this.config.secretKey)),
+      },
+      crypto.randomBytes(4).toString()
+    );
+    this.messaging = getMessaging(firebase);
+  }
+
+  async sendMessage(
+    options: IPushOptions
+  ): Promise<ISendMessageSuccessResponse> {
+    const res = await this.messaging.sendToDevice(options.target, {
+      notification: {
+        title: options.title,
+        body: options.content,
+        ...options.payload,
```

Comment:
I wonder if we should pass the payload object (I guess it's the same one from the trigger.event right?)

I think that it's better to provide a more strict interface here that will help us normalizing the communication across providers. The payload leaves a path to confusion and hard to understand what exactly should be there. Wdyt?

Reply:
I added stricter TypeScript typings. I think it is okay to do it like this, as user can specify the custom settings they need themselves. If there are any useless variables, firebase-admin will ignore them.

Reply:
I definitely aggree that it is indeed a very nice option to override the settings, but the issue without defining a more definitive object to facilitate them is that the user might be using some of the variables such as: color, or title in the context of a regular variable on his Email template, and it might create an un-expected side effect for the push notification.

If that's the case which I really love to provide this ability not only for FCM but other providers aswell, we can maybe create some other object in the trigger event other than the `payload`, maybe something like `overrides`? Wonder if that should be on the provider level, or just global one.